src/styles/text/text.js

```diff
@@ -92,15 +92,32 @@ Object.assign(TextStyle, {
             return;
         }
 
-        q.feature = feature;
-        q.context = context;
-        q.layout.vertex = false; // vertex placement option not applicable to standalone labels
+        // text can be an array if a `left` or `right` orientation key is defined for the text source
+        // in which case, push both text sources to the queue
+        if (q instanceof Array){
+            q.forEach(function(q){
+                q.feature = feature;
+                q.context = context;
+                q.layout.vertex = false; // vertex placement option not applicable to standalone labels
+
+                // Queue the feature for processing
+                if (!this.tile_data[tile.key]) {
+                    this.startData(tile);
+                }
+                this.queues[tile.key].push(q);
+            }.bind(this));
```

Comment:
can do. old habits die hard. what about arrow functions when `this` does not need to be bound?